sockeye/model.py

```diff
@@ -422,9 +426,12 @@ def load_parameters(self,
             utils.check_condition(not missing, f"missing keys: {missing}")
         if not ignore_extra:
             utils.check_condition(not unexpected, f"extra keys: {unexpected}")
-        # Models are saved with interleaved key-value params. If the current
-        # model is in training mode, separate the loaded params to match the
-        # format used during training.
+        # Models are saved with interleaved key-value params.
+        for module in self.modules():
+            if hasattr(module, 'kv_interleaved') and isinstance(module.kv_interleaved, bool):
```

Comment:
What is the check for bool for? Would we expect this member to ever be not bool?

Reply:
Right now, `kv_interleaved` is always `bool` if present. The check is for extra type safety and forward compatibility.

Reply:
this is a nice refactor, thanks